source/program/main.cpp

```diff
@@ -146,55 +146,141 @@ int multiworld_init(lua_State* L) {
+        std::vector<u8>* sendBuffer = new std::vector<u8>();               // sendBuffer to store the result. this pointer is returned 
 
         // +1; use lua's tostring so we properly convert all types
         lua_getglobal(L, "tostring");
 
         // +1
-        int loadResult = luaL_loadbuffer(L, buffer.data(), bufferLength, "remote lua");
+        int loadResult = luaL_loadbuffer(L, RecvBuffer.data() + 5, RecvBufferLength - 5, "remote lua");
 
         if (loadResult == 0) {
             // -1, +1 - call the code we just loaded
             int pcallResult = exefs_lua_pcall(L, 0, 1, 0);
             // -2, +1 - call tostring with the result of that
             lua_call(L, 1, 1);
 
-            size_t resultSize;
             const char* luaResult = lua_tolstring(L, 1, &resultSize);
             
             if (pcallResult == 0) {
                 // success! top string is the entire result
-                output_success = true;
-                memcpy(output_start, luaResult, std::min(output_buffer_size, resultSize));
-                response = resultSize;
-            } else {
-                // error happened
-                response = snprintf(output_start, output_buffer_size, "%s", luaResult);
-            }
+                outputSuccess = true;
+            } 
+            sendBuffer->insert(sendBuffer->begin(), luaResult, luaResult + resultSize);
         } else {
-            response = snprintf(output_start, output_buffer_size, "error parsing buffer: %d", loadResult);
+            std::string errorMessage = "error parsing buffer: " + std::to_string(loadResult);
+            const char* errorAsCString = errorMessage.c_str();
+            resultSize = strlen(errorAsCString);
+            sendBuffer->insert(sendBuffer->begin(), errorAsCString, errorAsCString + resultSize);
         }
-        buffer[0] = output_success;        
-        buffer[1] = response & 0xff;
-        buffer[2] = (response >> 8)  & 0xff;
-        buffer[3] = (response >> 16) & 0xff;
-        return response + 4;
+        sendBuffer->insert(sendBuffer->begin(), PACKET_REMOTE_LUA_EXEC);
+        sendBuffer->insert(sendBuffer->begin() + 1, outputSuccess);
+        sendBuffer->insert(sendBuffer->begin() + 2, resultSize & 0xff);
+        sendBuffer->insert(sendBuffer->begin() + 3, (resultSize >> 8)  & 0xff);
+        sendBuffer->insert(sendBuffer->begin() + 4, (resultSize >> 16)  & 0xff);
+        return sendBuffer;
     });
 
     // Register calling update again
     multiworld_schedule_update(L);
     return 0;
 }
 
+std::vector<u8>* create_packet_from_lua_string(lua_State* L) {
+    size_t resultSize = 0;          // length of the lua string response (without \0)
+    std::vector<u8>* sendBuffer;  // sendBuffer to store the result. this pointer is returned 
+    
+    const char* luaResult = lua_tolstring(L, 1, &resultSize);
+
+    sendBuffer = new std::vector<u8>();
+    if (sendBuffer == NULL) return NULL;

```

Comment:
By default allocator failures will come as exceptions in C++.
Not just that, but this allocator is just for the vector header - most allocations here are going to be happening inside the `insert` methods.

This means, all these guards for out of memory are doing nothing.

Reply:
Likely happened when switching from calloc to the std::vector.
Do we really want to handle these exceptions or don't care? If we don't have space for a few bytes to send a packet the game is probably doomed to run out of memory anyways.

Reply:
I don't mind just letting it blow up. Maybe we can wrap everything around a try catch somewhere at some point, if we ever have exceptions enabled

Reply:
I will just leave it for now without the out of memory exception handling. It would be a pain to debug this on ryujinx and console and is not worth the effort.